mutable_tree.go

```diff
@@ -268,7 +268,10 @@ func (tree *MutableTree) LazyLoadVersion(targetVersion int64) (int64, error) {
 		targetVersion = latestVersion
 	}
 
-	rootHash := tree.ndb.getRoot(targetVersion)
+	rootHash, err := tree.ndb.getRoot(targetVersion)
+	if err != nil {
+		return 0, err
```

Comment:
so here is the question: at which point we should retry DB operation (say 5 times) before bubbling up an error to the caller?

Reply:
in my view, I would prefer the application to handle this and here we present the error each time. @alexanderbez do you have a preference. The sdk is the main user of this repo so your preference matters most

Reply:
This change looks fine and behaves as I'd expect. If the version provided cannot be fetched, then the upstream caller should handle that error (either retry another version or fail completely).

Reply:
Agree. My question of course goes beyond this case.